boot/seal.go

```diff
@@ -95,7 +96,69 @@ func sealKeyToModeenv(key secboot.EncryptionKey, model *asserts.Model, modeenv *
 		return fmt.Errorf("cannot seal the encryption key: %v", err)
 	}
 
-	// TODO:UC20: store the predictable bootchains
+	// TODO:UC20: save the predictable bootchains
+
+	return nil
+}
+
+// resealKeyToModeenv reseals the existing encryption key to the
+// parameters specified in modeenv.
+func resealKeyToModeenv(model *asserts.Model, modeenv *Modeenv) error {
+	// build the recovery mode boot chain
+	rbl, err := bootloader.Find(InitramfsUbuntuSeedDir, &bootloader.Options{
+		Role: bootloader.RoleRecovery,
+	})
+	if err != nil {
+		return fmt.Errorf("cannot find the recovery bootloader: %v", err)
+	}
+
+	recoveryBootChains, err := recoveryBootChainsForSystems([]string{modeenv.RecoverySystem}, rbl, model, modeenv)
```

Comment:
here we should really use modeenv.CurrentRecoverySystems  (we undefine RecoverySystem after first seeding)

Reply:
Fixed. Thanks for spotting this, it was a result of the reseal function being largely based on the seal function.